lib/kubeclient/common.rb

```diff
@@ -257,16 +257,15 @@ def get_entities(entity_type, klass, resource_name, options = {})
       ns_prefix = build_namespace_prefix(options[:namespace])
       response = handle_exception do
         rest_client[ns_prefix + resource_name]
-        .get({ 'params' => params }.merge(@headers))
+          .get({ 'params' => params }.merge(@headers))
       end
 
       result = JSON.parse(response)
 
-      resource_version = result.fetch('resourceVersion', nil)
-      if resource_version.nil?
-        resource_version =
-            result.fetch('metadata', {}).fetch('resourceVersion', nil)
-      end
+      resource_version =
+        result.fetch('resourceVersion') do
+          result.fetch('metadata', {}).fetch('resourceVersion', nil)
+        end
```

Comment:
There is a mild difference:
given `result == {'resourceVersion' => nil, 'metadata' => {'resourceVersion' => 10}}`,
old code would take the 10, while new code returns nil — `fetch` runs its block only when the key is entirely absent.

Don't know if this matters (can both resourceVersion and metadata.resource be present but one be nil?), and don't know which behavior is better.
Anyway I think this would match old logic exactly:
```
resource_version = result.fetch('resourceVersion', nil)
resource_version ||= result.fetch('metadata', {}).fetch('resourceVersion', nil)
```
[Well not *exactly* exactly :-)  This still would still differ in handing of `false`, but it sounds safe to assume it's not a boolean.]